File: aricode/sparring/challenges/14_leibniz/leibniz.py
#!/usr/bin/env python3
# Python - Challenge 14: Leibniz Pi, return floor(pi*50)=157
import sys
s, sign, d = 0.0, 1.0, 1.0
for _ in range(10000):
    s += sign / d
    sign = -sign
    d += 2.0
sys.exit(int(s * 4.0 * 50.0))

File: aricode/sparring/challenges/14_leibniz/leibniz.c
/* C - Challenge 14: Leibniz Pi Approximation */
/* 10000 terms, return floor(pi * 50) = 157 */
int main(void) {
    double sum = 0.0, sign = 1.0, denom = 1.0;
    for (int i = 0; i < 10000; i++) {
        sum += sign / denom;
        sign = -sign;
        denom += 2.0;
    }
    return (int)(sum * 4.0 * 50.0);
}
